### PR TITLE
Updates for `fastparquet` evolution

### PR DESCRIPTION
**Pull request by martindurant**
- [ ] Closes #xxxx
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

Makes https://github.com/dask/fastparquet/pull/818 pass

### CODE REVIEW

**Review by phobson — APPROVED** [review]
Thanks, @martindurant

**Comment by martindurant** [comment]
@rjzamora : a quick and simple one, if you have a moment

File: dask/dataframe/io/tests/test_parquet.py [hunk 1]
```diff
@@ -3155,13 +3155,18 @@ def test_partitioned_column_overlap(tmpdir, engine, write_cols):
...
+        expect = pd.concat([_df1, _df2], ignore_index=True)
+        assert result.compute().reset_index(drop=True).to_dict() == expect.to_dict()
+    elif write_cols == ["part", "kind", "col"]:
         result = dd.read_parquet(path, engine=engine)
         expect = pd.concat([_df1, _df2], ignore_index=True)
         assert_eq(result, expect, check_index=False)
     else:
         # For now, partial overlap between partition columns and
-        # real columns is not allowed
+        # real columns is not allowed for pyarrow or older fastparquet
```
Comment (rjzamora) [inline dask/dataframe/io/tests/test_parquet.py:3169]: Should [the error message](https://github.com/dask/dask/blob/34a1e88bb3f6196361f398ddab55e59d315d8d40/dask/dataframe/io/parquet/fastparquet.py#L490) be changed to suggest a newer version of fastparquet?

Comment (martindurant) [reply dask/dataframe/io/tests/test_parquet.py:3169]: Amended

**Review by jrbourbeau — COMMENTED** [review]
Thanks @martindurant. I think we want to trigger an upstream CI build by including a commit with `test-upstream` in it so we can make sure `fastparquet_version > parse_version("0.8.3")` is `True` (note you'll also want to merge `main` as there were some changes to fix the `upstream` build that were just merged)

**Comment by martindurant** [comment]
There is no release of fastparquet yet, since https://github.com/dask/fastparquet/pull/818 is waiting on this PR

**Comment by jrbourbeau** [comment]
The upstream build will install `fastparquet` directly from `main`. I assumed this was using some new feature that's not yet released, but in `fastparquet`s `main` branch -- is that not the case?

**Comment by martindurant** [comment]
No, I didn't even merge it; but I will now, just to ease the impasse :). 
fastparquet also tests against main, of course!

**Comment by martindurant** [comment]
done

**Comment by martindurant** [comment]
dask/array/tests/test_stats.py::test_two[ttest_1samp-kwargs2] is nothing to do with this PR

**Pull request merged**